### PR TITLE
feat(web): implemented template images

### PR DESCRIPTION
**Pull request by robot9706**
Added the option to replace the image for templates. Each **JPG** image file should be **100x100** and should have the same name as the template json.

### CODE REVIEW

**Review by bvektor — APPROVED** [review]

File: web/crux-ui/locales/en/templates.json [hunk 1]
```diff
@@ -3,5 +3,6 @@
   "applyTemplate": "Create a Product from template '{{name}}'",
   "productName": "Product name",
   "description": "Description",
-  "type": "Type"
+  "type": "Type",
+  "imageAlt": "Template image"
```
Comment (m8vago) [inline web/crux-ui/locales/en/templates.json:7]: "templateImage": "Template image"

File: web/crux-ui/src/components/templates/template-image.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  const ref = useRef<HTMLImageElement>(null)
+  const [error, setError] = useState<boolean>(false)
+
+  const onError = useCallback(() => {
+    setError(true)
+  }, [])
+
+  useEffect(() => {
+    if (ref && ref.current) {
```
Comment (m8vago) [inline web/crux-ui/src/components/templates/template-image.tsx:23]: we should write a hook for these kind of errors

File: web/crux-ui/src/components/templates/template-image.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      if (errorLoadingImgBeforeHydration) {
+        setError(true)
+      }
+    }
+  }, [])
+
+  return error ? (
+    <Image src="/default_template.svg" width={100} height={100} />
+  ) : (
+    <img
```
Comment (m8vago) [inline web/crux-ui/src/components/templates/template-image.tsx:36]: use nextjs Image instead

File: web/crux-ui/src/pages/api/templates/[templateId]/image.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import crux from '@server/crux/crux'
+import { withMiddlewares } from '@server/middlewares'
+import { NextApiRequest, NextApiResponse } from 'next'
+
+const onGet = async (req: NextApiRequest, res: NextApiResponse) => {
+  const templateId = req.query.templateId as string
+
+  const data = await crux(req).templates.getImage(templateId)
+
+  res.status(200).setHeader('Content-Type', 'image/jpeg').send(data.data)
```
Comment (m8vago) [inline web/crux-ui/src/pages/api/templates/[templateId]/image.ts:10]: it think u can get rid of this whole endpoint with next13's custom imageLoader
https://nextjs.org/docs/api-reference/next/image

Comment (robot9706) [reply web/crux-ui/src/pages/api/templates/[templateId]/image.ts:10]: ImageLoader also solves the SSR error handling issue. 🚀

**Review by balanceee — APPROVED** [review]

**Review by m8vago — APPROVED** [review]

**Comment by polaroi8d** [comment]
Rebase please and merge.

**Pull request merged**